packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -744,6 +745,16 @@ abstract class AutoScalingGroupBase extends Resource implements IAutoScalingGrou
     });
   }
 
+  /**
+   * Add a pool of pre-initialized EC2 instances that sits alongside an Auto Scaling group
+   */
+  public addWarmPool(id: string, props: BasicWarmPoolProps): WarmPool {
```

Comment:
```suggestion
  public addWarmPool(options?: WarmPoolOptions): WarmPool {
```

Reply:
Instead of requiring an `id`, lets create it for the user instead. We do this all the time for the constructid elsewhere in the CDK.

Reply:
Sounds good. Will update in the next revision.